corehq/apps/userreports/templates/userreports/edit_data_source.html

```diff
@@ -31,5 +31,10 @@
     <a href="{% url 'preview_configurable_data_source' domain data_source.get_id %}" class="btn">{% trans 'Preview data' %}</a>
     <a href="{% url 'configurable_data_source_json' domain data_source.get_id %}" class="btn">{% trans 'Data Source Source (Advanced)' %}</a>
 </form>
+{% if not data_source.is_deactivated %}
```

Comment:
I'd be down to look at it in the near future (perfect task for being on the train).